front/lib/api/assistant/conversation.ts

```diff
@@ -1120,270 +1121,302 @@ export async function* editUserMessage(
     return;
   }
 
-  // In one big transaction creante all Message, UserMessage, AgentMessage and Mention rows.
-  const { userMessage, agentMessages, agentMessageRows } =
-    await front_sequelize.transaction(async (t) => {
-      await getConversationRankVersionLock(conversation, t);
+  // local error class to differentiate from other errors
+  class UserMessageError extends Error {}
 
-      const messageRow = await Message.findOne({
-        where: {
-          sId: message.sId,
-          conversationId: conversation.id,
-        },
-        include: [
-          {
-            model: UserMessage,
-            as: "userMessage",
-            required: true,
+  try {
+    // In one big transaction creante all Message, UserMessage, AgentMessage and Mention rows.
+    const { userMessage, agentMessages, agentMessageRows } =
+      await front_sequelize.transaction(async (t) => {
+        await getConversationRankVersionLock(conversation, t);
+
+        const messageRow = await Message.findOne({
+          where: {
+            sId: message.sId,
+            conversationId: conversation.id,
           },
-        ],
-      });
-      if (!messageRow || !messageRow.userMessage) {
-        throw new Error(
-          "Unexpected: Message or UserMessage to edit not found in DB"
-        );
-      }
-      const userMessageRow = messageRow.userMessage;
-      // adding messageRow as param otherwise Ts doesn't get it can't be null
-      async function createMessageAndUserMessage(messageRow: Message) {
-        return await Message.create(
-          {
-            sId: generateModelSId(),
+          include: [
+            {
+              model: UserMessage,
+              as: "userMessage",
+              required: true,
+            },
+          ],
+        });
+        if (!messageRow || !messageRow.userMessage) {
+          throw new Error(
+            "Unexpected: Message or UserMessage to edit not found in DB"
+          );
+        }
+        const newerMessage = await Message.findOne({
+          where: {
             rank: messageRow.rank,
             conversationId: conversation.id,
-            parentId: messageRow.parentId,
             version: messageRow.version + 1,
-            userMessageId: (
-              await UserMessage.create(
+          },
+        });
+        if (newerMessage) {
+          throw new UserMessageError(
+            "Unexpected: Message to edit is not the latest version"
```

Comment:
The message can be better:

Invalid user message edit request, this message was already edited.